processor/README.md

```diff
@@ -254,22 +289,101 @@ Refer to [config.yaml](batchprocessor/testdata/config.yaml) for detailed
+memory_limiter processor on every collector. The ballast should be configured to
+be 1/3 to 1/2 of the memory allocated to the collector. The memory_limiter
+processor should be the first processor defined in the pipeline (immediately after
+the receivers). This is to ensure that backpressure can be sent to applicable
+receivers and minimize the likelihood of dropped data when the memory_limiter gets
+triggered.
+
+Please refer to [config.go](memorylimiter/config.go) for the config spec.
+
+The following configuration options **must be changed**:
+- `check_interval` (default = 0s): Time between measurements of memory
+usage. Values below 1 second are not recommended since it can result in
+unnecessary CPU consumption.
+- `limit_mib` (default = 0): Maximum amount of memory, in MiB, targeted to be
+allocated by the process heap. Note that typically the total memory usage of
+process will be about 50MiB higher than this value.
+- `spike_limit_mib` (default = 0): Maximum spike expected between the
+measurements of memory usage. The value must be less than `limit_mib`.
+
+The following configuration options can also be modified:
+- `ballast_size_mib` (default = 0): Must match the `mem-ballast-size-mib`
+command line option.
+
+Examples:
+
+```yaml
+processors:
+  memory_limiter:
+    ballast_size_mib: 2000
+    check_interval: 5s
+    limit_mib: 4000
+    spike_limit_mib: 500
+```
+
+Refer to [config.yaml](memorylimiter/testdata/config.yaml) for detailed
+examples on using the processor.
+
+
+## <a name="queued-retry"></a>Queued Retry Processor
 
-The queued_retry processor uses a bounded queue to relay trace data from the receiver
-or previous processor to the next processor. Received trace data is enqueued
-immediately if the queue is not full. At the same time, the processor has one
-or more workers which consume the trace data in the queue by sending them to
-the next processor. If relaying the trace data to the next processor or
-exporter in the pipeline fails, the processor retries after some backoff delay
-depending on the configuration (see below). Please refer to
-[config.go](queuedprocessor/config.go) for the config spec.
+The queued_retry processor uses a bounded queue to relay batches from the receiver
+or previous processor to the next processor. Received data is enqueued immediately
+if the queue is not full. At the same time, the processor has one or more workers
+which consume the data in the queue by sending them to the next processor or exporter.
+If relaying the data to the next processor or exporter in the pipeline fails, the
+processor retries after some backoff delay depending on the configuration.
+
+Given that if the queue is full the data will be dropped, it is important to size
+the queue appropriately. The queue is kept in memory today, so the larger the queue
+the more memory consumed by the collector that could otherwise be used for
+other purposes. Also, since the queue is based on batches and batch sizes are
```

Comment:
The queue is pay-for-play, so if there is very little actual queueing there is no memory penalty. The memory used is proportional to the actual number of items in the queue. I mean a queue of "max size" 1 or 100K will have the same footprint if there is a single batch on the queue.